go/adbc/driver/databricks/record_builder.go

```diff
@@ -0,0 +1,338 @@
+		builder: array.NewBooleanBuilder(mem),
+	}
+}
+
+func (b *BooleanColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	if v, ok := value.(bool); ok {
+		b.builder.Append(v)
+	} else {
+		return fmt.Errorf("invalid type for Boolean column: %T", value)
+	}
+	return nil
+}
+
+func (b *BooleanColumnBuilder) Builder() array.Builder {
+	return b.builder
+}
+
+// TimestampColumnBuilder handles timestamp values
+type TimestampColumnBuilder struct {
+	builder *array.TimestampBuilder
+}
+
+func NewTimestampColumnBuilder(mem memory.Allocator) *TimestampColumnBuilder {
+	return &TimestampColumnBuilder{
+		builder: array.NewTimestampBuilder(mem, &arrow.TimestampType{Unit: arrow.Microsecond}),
+	}
+}
+
+func (b *TimestampColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	if t, ok := value.(time.Time); ok {
+		b.builder.Append(arrow.Timestamp(t.UnixMicro()))
+	} else {
+		return fmt.Errorf("invalid type for Timestamp column: %T", value)
+	}
+	return nil
+}
+
+func (b *TimestampColumnBuilder) Builder() array.Builder {
+	return b.builder
+}
+
+// Date32ColumnBuilder handles date32 values
+type Date32ColumnBuilder struct {
+	builder *array.Date32Builder
+}
+
+func NewDate32ColumnBuilder(mem memory.Allocator) *Date32ColumnBuilder {
+	return &Date32ColumnBuilder{
+		builder: array.NewDate32Builder(mem),
+	}
+}
+
+func (b *Date32ColumnBuilder) Append(value interface{}) error {
+	if value == nil {
+		b.builder.AppendNull()
+		return nil
+	}
+	if t, ok := value.(time.Time); ok {
+		b.builder.Append(arrow.Date32(t.Unix() / 86400))
+	} else {
+		return fmt.Errorf("invalid type for Date32 column: %T", value)
+	}
+	return nil
+}
+
+func (b *Date32ColumnBuilder) Builder() array.Builder {
+	return b.builder
+}
+
+// RecordBuilder handles building Arrow records from arrays of data
+type RecordBuilder struct {
+	schema   *arrow.Schema
+	mem      memory.Allocator
+	builders []ColumnBuilder
+}
+
+// NewRecordBuilder creates a new RecordBuilder with the given schema
+func NewRecordBuilder(schema *arrow.Schema) *RecordBuilder {
+	mem := memory.NewGoAllocator()
+	builders := make([]ColumnBuilder, len(schema.Fields()))
+
+	for i, field := range schema.Fields() {
+		switch field.Type.ID() {
+		case arrow.STRING:
+			builders[i] = NewStringColumnBuilder(mem)
+		case arrow.INT32:
+			builders[i] = NewInt32ColumnBuilder(mem)
+		case arrow.INT64:
+			builders[i] = NewInt64ColumnBuilder(mem)
+		case arrow.FLOAT32:
+			builders[i] = NewFloat32ColumnBuilder(mem)
+		case arrow.FLOAT64:
+			builders[i] = NewFloat64ColumnBuilder(mem)
+		case arrow.BOOL:
+			builders[i] = NewBooleanColumnBuilder(mem)
+		case arrow.TIMESTAMP:
+			builders[i] = NewTimestampColumnBuilder(mem)
+		case arrow.DATE32:
+			builders[i] = NewDate32ColumnBuilder(mem)
+		default:
+			// Default to string for unknown types
+			builders[i] = NewStringColumnBuilder(mem)
```

Comment:
We should panic here instead as this is only reachable if we don't match the schema derivation function with this one correctly.